app/src/main/java/com/globant/practice/domain/interactor/UserInteractor.java

```diff
@@ -0,0 +1,92 @@
+package com.globant.practice.domain.interactor;
+
+import com.globant.practice.domain.model.User;
+import com.globant.practice.domain.service.ProvideGithubApi;
+import java.util.List;
+import javax.inject.Inject;
+import io.reactivex.Observable;
+import io.reactivex.Observer;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+
+/**
+ * Provides all the methods necessary to manage the user information.
+ * Created by jonathan.vargas on 4/04/2017.
+ */
+
+public class UserInteractor {
+
+    private ProvideGithubApi api;
+    private List<User> usersList;
+    private Observable<List<User>> getUsersListObservable;
+
+    /**
+     * Construct method of UserInteractor that receives a GithubApi reference.
+     *
+     * @param api
+     */
+    @Inject
+    public UserInteractor(ProvideGithubApi api) {
+        this.api = api;
+    }
+
+    /**
+     * Returns a Observable reference that can be a reference of the api call o a reference of the
+     * list depending if the list is null or not.
+     *
+     * @return
+     */
+    public Observable<List<User>> getUsers() {
+        if (usersList == null) {
+            getUsersListObservable = api.getUsers();
+            getUsersListObservable.subscribeOn(Schedulers.newThread())
```

Comment:
You're already subscribing in the presenter, there is no need to duplicate the logic